.pre-commit-config.yaml

```diff
@@ -0,0 +1,13 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+
+# Update with: pre-commit autoupdate --freeze
+
+exclude: '_vendor'
+
+repos:
+-   repo: https://github.com/asottile/pyupgrade
+    rev: e695ecd365119ab4e5463f6e49bea5f4b7ca786b  # frozen: v2.31.0
+    hooks:
+    -   id: pyupgrade
+        args: [--py37-plus]
```

Comment:
Normally, I like to manage changes like these in [jaraco/skeleton](/jaraco/skeleton). I'm not confident, however, that this change could be accepted there. Is this change a best practice or a setuptools-specific change? How about let's treat this concern (a pre-commit hook) separately from the code changes?

Reply:
I'm not sure why the author pinned this tool to an old version, but I at least don't consider that to be a best practice. Or do you mean using pyupgrade (in pre-commit) in general? That I do in all of my new repositories.

Reply:
I mean pyupgrade and pre-commit in general. In the skeleton, I've added 'black' as a pre-commit hook, but I removed blacken docs (https://github.com/jaraco/skeleton/commit/719a7ced8a1713b7fe94d842a8f6fec7425b8a0a). Last night, I explored the possibility of using pyupgrade, but found it [doesn't support compatible changes](https://github.com/asottile/pyupgrade/issues/659). Hence, I'm reluctant to apply it as a best practice or even in this project. I don't want to enforce a behavior that's known unsafe and hostile.

Reply:
In other projects, I've started applying pyupgrade at strategic times and not used any pre-commit hook. Let's abandon that concern for now and focus on simply running the upgrade.